### PR TITLE
29173 popup

### PR DESCRIPTION
**Pull request by AlejandroColinaGSA**

### CODE REVIEW

**Comment by AlejandroColinaGSA** [comment]
please see:
https://cg-06ab120d-836f-49a2-bc22-9dfb1585c3c6.app.cloud.gov/preview/gsa/sam-styles/29173-popup/components/detail/icons.html
Screen Shot 2020-01-13 at 1 09 09 PM

https://cg-06ab120d-836f-49a2-bc22-9dfb1585c3c6.app.cloud.gov/preview/gsa/sam-styles/29173-popup/components/detail/popup.html
Screen Shot 2020-01-13 at 1 10 52 PM

File: src/stylesheets/components/_results.scss [hunk 1]
```diff
@@ -9,4 +9,29 @@
     justify-content: center;
     flex-direction: column;
   }
-}
+
+  &.dual-record {
+    background-color: #f9f9f9;
```
Comment (davereed) [inline src/stylesheets/components/_results.scss]: we need to use one of our base colors rather than hard code a hex

Comment (davereed) [reply src/stylesheets/components/_results.scss]: Use #f9f9f7

File: src/components/08-miscellaneous/07-popup/popup.njk [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<div class="sds-popup">
+  <i class="sds fa-information-circle fa-2x"></i>
+  <div class="sds-popup__content">
+    <i class="sds fa-information-circle fa-2x float-right"></i>
```
[the remark this answers is not in the document]
Comment (AlejandroColinaGSA) [reply src/components/08-miscellaneous/07-popup/popup.njk]: addressed

File: src/stylesheets/components/_results.scss (hunk 1 above)
Comment (AlejandroColinaGSA) [reply src/stylesheets/components/_results.scss]: changed to #f9f9f7 (per Kapil)

File: src/stylesheets/components/_results.scss [hunk 3]
```diff
@@ -9,4 +9,29 @@
     justify-content: center;
     flex-direction: column;
   }
-}
+
```
Comment (davereed) [inline src/stylesheets/components/_results.scss]: This should be in a separate PR, it's not related to popup component.

File: src/stylesheets/components/_popup.scss [hunk 4]
```diff
@@ -0,0 +1,64 @@
+  .sds-popup {
+      position: relative;
```
[the remark this answers is not in the document]
Comment (AlejandroColinaGSA) [reply src/stylesheets/components/_popup.scss]: the issue is where the pop-up content box appears, perhaps we build it as sds-popup__left and sds-popup__right, we would then provide a parameter in the component.

File: src/stylesheets/components/_results.scss (hunk 3 above)
Comment (AlejandroColinaGSA) [reply src/stylesheets/components/_results.scss]: I removed this

File: src/stylesheets/components/_popup.scss [hunk 5]
```diff
@@ -0,0 +1,68 @@
+.sds-popup {
+  position: relative;
+  display:initial;
+}
+.sds-popup__content {
+  //outline: 1px solid red;
+}
```
Comment (davereed) [inline src/stylesheets/components/_popup.scss]: Remove commented code

File: src/stylesheets/components/_popup.scss [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    &.top-right     {left: 0; bottom: 100%; right: 100%; text-align: end;}
+    &.right-middle  {left: 0; top: 0; right: 100%; text-align: end; height:100%;
+            *:first-child {position: absolute;  top: 50%;  transform: translateY(-50%);
+    }}
+    &.bottom-right  {top: 100%;  left: 0;  text-align: right;}
+    &.bottom-center {top: 100%;  left: 0;  text-align: center}
+    &.bottom-left   {top: 100%;  left: 0;  text-align: left}
+    &.left-middle   {left: 0; top: 0; right: 0%; text-align: start; height:100%;
+      *:first-child {position: absolute;  top: 50%; right:100%; transform: translateY(-50%);
+        /* beautify preserve:end */
```
Comment (davereed) [inline src/stylesheets/components/_popup.scss]: is this in the right spot?

File: src/stylesheets/components/_popup.scss [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+      }
+    }
+    .tooltip {
+      background-color: color("base-dark");
+      border: 1px solid color("base-darkest");
+      @include u-text("base-lightest");
+      @include u-font("sans", "3xs");
+      padding: 2px 5px;
```
Comment (davereed) [inline src/stylesheets/components/_popup.scss]: Can we use preset values?

File: src/stylesheets/components/_popup.scss [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+      white-space: nowrap;
+      display: inline-block;
+      @include u-margin(1px);
```
Comment (davereed) [inline src/stylesheets/components/_popup.scss]: This should be a string value for "1px"

File: src/stylesheets/components/_popup.scss [hunk 9]
```diff
@@ -0,0 +1,67 @@
+.sds-popup {
+  position: relative;
+  display:inline-block;
+}
+.sds-popup__content {
+}
```
Comment (davereed) [inline src/stylesheets/components/_popup.scss]: Remove the unused class

File: src/components/08-miscellaneous/07-popup/popup.njk [hunk 10]
```diff
@@ -0,0 +1,213 @@
...
+                          <p>The Second item</p><hr class="thin"/>
+                          <p>Another item</p><hr class="thin"/>
+                          <p>One More item</p><hr class="thin"/>
+                          <p>The Last item</p>
+
+                      </div>
+                    </div>
+                  </div>
+                </div>
+                <div class="tablet:grid-col-auto text-right sds-popup">
```
Comment (davereed) [inline src/components/08-miscellaneous/07-popup/popup.njk]: **For all popups** please refer to the guidelines for accessibility:
https://www.w3.org/TR/wai-aria-1.2/#tooltip
http://accessibility.athena-ict.com/aria/examples/tooltip.shtml


Main concern is that they don't appear on focus, missing role attribute, and missing aria-describeby.
